### DISCUSSION TITLE
[PROPOSAL] CHOICE Balancing Land Expansion, Choosing your Specialty or Balance

### DISCUSSION THREAD

**Discussion by JustJayHSP** [sunflower-land/sunflower-land#2293] [Ideas]
We already know that whenever you Expand, you have a chance to get the best, above average, average, below average, worst set pieces.

After playing Sunflower Isles I enjoyed it very much but this is a bad RNG system. Comparing myself to the best or even just above average player. They have better Crop Plots, Trees, Stones, Iron and Gold than the Worst Luck farmer. 
Which is in my honest opinion having only a quantity of 74 while the best having a 92 quantity, is in the end very depressing.
You have like 5% chance more or less to get the best plot. In my experience I have been getting below average and worst luck when expanding. I'm just 1 Quantity above the worst farm. (75 Quantity)

**COMPARISON**
Best Max Expansion: 46 Plots, 13 Trees, 11 Stone, 7 Iron, 7 Gold (92 Quantity)
Worst Max Expansion: 38 Plots, 12 Trees, 9 Stone, 4 Iron, 3 Gold (74 Quantity) **[A Large Gap of 18 Entities]**

What I'm going to suggest is that there should be a choice every time you expand and there should always be a Standard
A STANDARD/MINIMUM of 38 Plots, 10 Trees, 8 Stone, 4 Iron, 3 Gold + Your Choice.

![Bonus Expansion Slots](https://user-images.githubusercontent.com/107456378/221597199-cdb1c9aa-ffe4-4b52-8c0f-9be96913c7d8.jpg)

What I mean by this choice is, whenever you expand your land, a prompt shows up 
A. Do you want more Crop Plots?
B. More Trees?
C. More Stone?
D. More Iron?
E. More Gold?
F. More Fruit Trees?
G. More Space?

After selecting, the system then adds in that Choice Entity to your Farm. 

With of course we need some kind of **Maximum Limit of certain Nodes/Plots** say like, you can only have 7 Iron Nodes or 5 Gold Nodes.

EDITED:
On your next expansion the option for extra Iron Nodes is removed because you maximized the amount of Iron Nodes for a farm.
We can call it a CEILING or a MAXIMUM specific Entity.
You can only have
38 to 43 Plots
10 to 13 Trees
8 to 11 Stone
4 to 7 Iron
3 to 5 Gold    (You wont get everything here only what you upgraded)       

**YOUR CHOICE MATTERS ON WHAT YOU WANT TO IMPROVE AND THERE WILL ALWAYS BE A CEILING or LIMIT**  
You won't get every Entity maximized, there will be a choice

This will absolutely render the wide gap of Entities totally null and void from the Average Farmer to the Best Farmer. Having the same Total Number of Entities with their own preferred choice.

This in turn will make sure that there are players that have some sort Specialty or Balance if they wanted.
Specialty that could focus on what they want instead of praying to god you have enough Iron Nodes and Stone to keep mining Gold.

You could be a Stone Miner Specialty, a Tree Chopper or a Farmer Specialty.
or a Balance that they could be a Jack of All trades but a master of none. 

Instead of having a System where you could have the worst imaginable Farm or the best farm imaginable. 
We all have a choice....and its not some RNG based Farm Expansion mechanic where you pray that your Farm is good and if its not, you're out of luck not unless you have $200-400 to buy a new Farm that has the best expansions?

I know there might be another similar discussion but I still would like to emphasize on this kind of CHOICE in Land Expansion since comparing a Lvl 16 Land Expansion from the Best and the Worst or even an Average farm has a wide gap of 18 Entities.

**After the Lvl 16th Land Expansion**
We can focus more on additional options

A. Would you like more Boulders?
B. Would you like more Fruit Trees?
C. Would you like more Fishing Spots?
D. Would you like more Wild Animal Spawn Areas?

In the end, this is a Balancing Fix that has been complained over a few months back and of course the Developer's have the right to adjust these values. This is This and That's That.

**Comment by Aeon-LlamaMod** [comment]
I think this is a good start. My issue is that players will just choose the high end resources to get more out of selling to others. This would devalue the end game resources such as gold/iron/boulders etc.

Now if you limit that to only a few expansions and your choice was extra plots or extra trees and stones maybe that would work better but it would have to be made so that the choice is not available all the time.

As plots are the main means to generate SFL many would just choose plots and get 100 plots to grow crops which would inflate SFL and deflate the other resources. 

All in all i think its definitely a right step into allowing player choice and to make things more balanced for others. Good work!

**Reply by JustJayHSP** [reply]
Thank you for responding and providing input. Yes I only desire is a step up towards the unbalanced expansions and not a 100% implementation. I'm sure my idea will get re-adjusted and yes I do hope there's an Upper Limit or a Ceiling to what you can have.

**Reply by JustJayHSP** [reply]
As you said people will just prioritize on getting more crops but that doesn't mean they would get 100 Plots.

If the best right now is 46 Plots and the Worst is 38 Plots then we could meet in the above average middle of 42/43 plots as the maximum allowed plots at the moment if they prioritized on it.

I also think that if you want to maximize a certain aspect on your island be sure that there will be an increase of Materials just to add that sort of resource

Will edit the data to emphasize more on the Ceiling aspect

**Reply by SuperSynapse** [reply]
@JustJayHSP 

This is a good basic concept. IMO it's focused too much on patching the current issue, not the underlying problem.

Some suggestions:
- What is the plan/proposal for the intended gameplay of players choosing different nodes? Is the intent to open up more trade, or to allow players to better balance their farm? What are the consequences of these actions?

- How will choosing nodes/plots affect gameplay? How will specialization affect gameplay? How will players exploit it and how will this design counter those imbalances?

- What is the proposed value of these nodes? Can I choose 1 plot or 1 gold node? or do they have different cost per land expansion?

- Why is RNG less desirable? What are the underlying reasons players should choose rather than it being chosen for them?

- What are the proposed minimums for plots/nodes, and proposed maximums? What is the thought behind why you chose these numbers? (some of this you briefly touched on).

Realize you're making changes to a machine in motion, not giving it a new coat of paint. There will be consequences for any changes better or worse, and you need to be ahead of how players will use them to take advantage of the new system.

I touched on some of the game theory behind choosing paths in my post. May be worthwhile to take ideas from, or talk about things you don't like in this post:

https://github.com/sunflower-land/sunflower-land/discussions/2253

Dcol also is making a skill tree, how does this affect your idea?
https://github.com/sunflower-land/sunflower-land/discussions/2180

How about c-py's post on Mutiple bumpkins?
https://github.com/sunflower-land/sunflower-land/discussions/2211

<hr>

@Aeon-LlamaMod 
> My issue is that players will just choose the high end resources to get more out of selling to others. 

This would be good. We see the value of gold go down relative to other resources people don't choose such as wood, stone, animals, crops etc.

The economy of choice.  If everyone goes one direction, they open up better paths for others, and for new players to capitalize on through trade. IMO having different paths makes ways for everyone to succeed and strategize without every advancement being a "pre made choice" or a "obvious linear advancement".

**Comment by TheDcol** [comment]
This could be an interesting approach, especially if there was a reset of expansions (refunds of course) and everyone started again (unlikely to happen, but we can hypothesize)

You could create some sort of control whilst giving players options..

**Comment by rasheed19871** [comment]
this is a very good suggestion at least we have some control toward our main goal
